src/coap_mbedtls.c

```diff
@@ -0,0 +1,1784 @@
+/*
+* coap_mbedtls.c -- mbedTLS Datagram Transport Layer Support for libcoap
+*
+* Copyright (C) 2019 Jon Shallow <supjps-libcoap@jpshallow.com>
+*               2019 Jitin George <jitin@espressif.com>
+*
+* This file is part of the CoAP library libcoap. Please see README for terms
+* of use.
+*/
+
+/*
+ * Naming used to prevent confusion between coap sessions, mbedtls sessions etc.
+ * when reading the code.
+ *
+ * c_context  A coap_context_t *
+ * c_session  A coap_session_t *
+ * m_context  A coap_mbedtls_context_t * (held in c_context->dtls_context)
+ * m_env      A coap_mbedtls_env_t * (held in c_session->tls)
+ */
+
+#include "coap_internal.h"
+
+#ifdef HAVE_MBEDTLS
+
+/*
+ * This code can be conditionally compiled to remove some components if
+ * they are not required to make a lighter footprint - all based on how
+ * the mbedtls library has been built.  These are not defined within the
+ * libcoap environment.
+ *
+ * MBEDTLS_SSL_SRV_C - defined for server side functionality
+ * MBEDTLS_SSL_CLI_C - defined for client side functionality
+ * MBEDTLS_SSL_PROTO_DTLS - defined for DTLS support
+ * MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED - defined if PSK is to be supported
+ *
+ * Note: TLS is not currently supported until additional code is added
+ */
+
+#include <mbedtls/version.h>
+#include <mbedtls/platform.h>
+#include <mbedtls/net_sockets.h>
+#include <mbedtls/ssl.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/error.h>
+#include <mbedtls/certs.h>
+#include <mbedtls/timing.h>
+#include <mbedtls/ssl_cookie.h>
+#include <mbedtls/oid.h>
+#include <mbedtls/debug.h>
+#if defined(ESPIDF_VERSION) && defined(CONFIG_MBEDTLS_DEBUG)
+#include <mbedtls/esp_debug.h>
+#endif /* ESPIDF_VERSION && CONFIG_MBEDTLS_DEBUG */
+#include <errno.h>
+#include <arpa/inet.h>
```

Comment:
This header file is not required anymore.

Reply:
Removed.